std/math.d

```diff
@@ -6883,6 +6883,44 @@ debug(UnitTest)
 real nextUp(real x) @trusted pure nothrow @nogc
 {
     alias F = floatTraits!(real);
+    static if (F.realFormat != RealFormat.ieeeDouble)
+    if (__ctfe)
+    {
+        if (x == -real.infinity)
+            return -real.max;
+        if (!(x < real.infinity)) // Infinity or NaN.
+            return x;
+        real delta;
+        // Start with a decent estimate of delta.
+        if (x <= 0x1.ffffffffffffep+1023 && x >= -double.max)
+        {
+            const double d = cast(double) x;
+            static if (real.mant_dig >= 64)
+                delta = (cast(real) nextUp(d) - cast(real) d) * 0x1p-11L;
+            else // e.g. RealFormat.ieeeExtended53
+                delta = cast(real) nextUp(d) - cast(real) d;
```

Comment:
Doesn't look like this is necessary for 80-bit reals with 53-bit precision, you can just use the first branch unconditionally.